application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarksSystem.java

```diff
@@ -0,0 +1,185 @@
+    public static final int WARN_BOOKMARK_COUNT_TOTAL = 900_000;
+    public static final int MAX_BOOKMARK_COUNT_USER = 500;
+    public static final int MAX_NOTE_LENGTH = 150;
+    public static final Duration LEAVE_BOOKMARKS_REMOVAL_DELAY = Duration.ofDays(7);
+
+    private final Database database;
+    private final Predicate<String> isOverviewChannelName;
+
+    private Consumer<GenericCommandInteractionEvent> listPaginationConsumer = null;
+    private Consumer<GenericCommandInteractionEvent> removePaginationConsumer = null;
+
+    /**
+     * Creates a new instance of the bookmarks system.
+     *
+     * @param config The config to use
+     * @param database The database to use
+     */
+    public BookmarksSystem(Config config, Database database) {
+        this.database = database;
+
+        isOverviewChannelName = Pattern.compile(config.getHelpSystem().getOverviewChannelPattern())
+            .asMatchPredicate();
+    }
+
+    /**
+     * Accepts the consumers for used for requesting the list and remove paginations.
+     *
+     * @param listPaginationConsumer The list pagination handler
+     * @param removePaginationConsumer The remove pagination handler
+     */
+    void acceptPaginationConsumers(Consumer<GenericCommandInteractionEvent> listPaginationConsumer,
+            Consumer<GenericCommandInteractionEvent> removePaginationConsumer) {
+        this.listPaginationConsumer = listPaginationConsumer;
+        this.removePaginationConsumer = removePaginationConsumer;
+    }
+
+    /**
+     * Requests a list pagination from the {@link BookmarksPaginatorInteractor}.
+     *
+     * @param event The command interaction event
+     */
+    void requestListPagination(GenericCommandInteractionEvent event) {
+        Objects.requireNonNull(listPaginationConsumer, "No list pagination consumer was provided");
+
+        listPaginationConsumer.accept(event);
+    }
+
+    /**
+     * Requests a remove pagination from the {@link BookmarksPaginatorInteractor}.
+     *
+     * @param event The command interaction event
+     */
+    void requestRemovePagination(GenericCommandInteractionEvent event) {
+        Objects.requireNonNull(removePaginationConsumer,
+                "No remove pagination consumer was provided");
+
+        removePaginationConsumer.accept(event);
+    }
+
+    boolean isHelpThread(MessageChannelUnion channel) {
+        if (channel.getType() != ChannelType.GUILD_PUBLIC_THREAD) {
+            return false;
+        }
+
+        ThreadChannel threadChannel = channel.asThreadChannel();
+        String parentChannelName = threadChannel.getParentMessageChannel().getName();
+
+        return isOverviewChannelName.test(parentChannelName);
+    }
+
+    boolean didUserBookmarkChannel(long userID, long channelID) {
+        return database.read(context -> context.selectFrom(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(userID), BOOKMARKS.CHANNEL_ID.eq(channelID))
+            .fetch()
+            .isNotEmpty());
```

Comment:
right now ur DB query retrieves EVERYTHING. instead, add a `limit(1)` to the query. `.limit(1).fetchOne() != null` or sth like that